public/app/features/explore/ExploreGraphNGPanel.tsx

```diff
@@ -0,0 +1,180 @@
+import {
+  AbsoluteTimeRange,
+  applyFieldOverrides,
+  DataFrame,
+  dateTime,
+  Field,
+  FieldColorModeId,
+  GrafanaTheme,
+  TimeZone,
+  TIME_SERIES_VALUE_FIELD_NAME,
+} from '@grafana/data';
+import {
+  Collapse,
+  DrawStyle,
+  GraphNG,
+  GraphNGLegendEvent,
+  Icon,
+  LegendDisplayMode,
+  TooltipPlugin,
+  useStyles,
+  useTheme,
+  ZoomPlugin,
+} from '@grafana/ui';
+import { hideSeriesConfigFactory } from 'app/plugins/panel/timeseries/hideSeriesConfigFactory';
+import { ContextMenuPlugin } from 'app/plugins/panel/timeseries/plugins/ContextMenuPlugin';
+import { ExemplarsPlugin } from 'app/plugins/panel/timeseries/plugins/ExemplarsPlugin';
+import { css, cx } from 'emotion';
+import React, { useCallback, useEffect, useState } from 'react';
+import { splitOpen } from './state/main';
+import { getFieldLinksForExplore } from './utils/links';
+
+const MAX_NUMBER_OF_TIME_SERIES = 20;
+
+interface Props {
+  data: DataFrame[];
+  annotations?: DataFrame[];
+  isLoading: boolean;
+  width: number;
+  absoluteRange: AbsoluteTimeRange;
+  timeZone: TimeZone;
+  onUpdateTimeRange: (absoluteRange: AbsoluteTimeRange) => void;
+  splitOpenFn: typeof splitOpen;
+}
+
+export function ExploreGraphNGPanel({
+  width,
+  data,
+  timeZone,
+  absoluteRange,
+  onUpdateTimeRange,
+  isLoading,
+  annotations,
+  splitOpenFn,
+}: Props) {
+  const [showAllTimeSeries, setShowAllTimeSeries] = useState(false);
+  const theme = useTheme();
+  const [dataFramesWithConfig, setDataFramesWithConfig] = useState<DataFrame[]>();
+  const style = useStyles(getStyles);
+  const timeRange = {
+    from: dateTime(absoluteRange.from),
+    to: dateTime(absoluteRange.to),
+    raw: {
+      from: dateTime(absoluteRange.from),
+      to: dateTime(absoluteRange.to),
+    },
+  };
+
+  const onLegendClick = useCallback(
+    (event: GraphNGLegendEvent) => {
+      const configWithOverrides = hideSeriesConfigFactory(event, { defaults: fieldConfig, overrides: [] }, data);
+      const applied = applyFieldOverrides({
+        fieldConfig: configWithOverrides,
+        data,
+        timeZone,
+        replaceVariables: value => value, // TODO: replace with real replace
```

Comment:
`/ / TODO: replace with real replace` What does this mean? Is it for template variables? Cause we don't have any in Explore.

Reply:
You are right. We don't need this.